### PR TITLE
MONIT-29809 changes to ingested "error_name" instead of "exception" for summary

### PR DESCRIPTION
**Pull request by panjwanipr-vmware**
[java-lib]changes to ingested "error_name" instead of "exception" for summary

### CODE REVIEW

File: java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java [hunk 1]
```diff
@@ -674,12 +676,88 @@ public static List<String> getDefaultLogMessageKeys() {
     return DEFAULT_LOG_MESSAGE_KEYS;
   }
 
+  @Nullable
+  public static String getLogLevel(@Nullable List<Annotation> annotations,
+                                     @Nullable List<String> customLogLevelTags) {
+    String serviceStr = null;
```
Comment (xuranchen) [inline java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java:682]: nit: serviceStr -> logLvlStr

**Review by xuranchen — DISMISSED** [review]
one nit, otherwise LGTM

File: java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java (hunk 1 above)
Comment (panjwanipr-vmware) [reply java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java]: Done change.

**Review by kabathla — DISMISSED** [review]

File: java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java [hunk 2]
```diff
@@ -674,12 +676,88 @@ public static List<String> getDefaultLogMessageKeys() {
...
 
+  @Nullable
+  public static String getLogLevel(@Nullable List<Annotation> annotations,
+                                     @Nullable List<String> customLogLevelTags) {
+    String logLvlStr = null;
+    if (annotations != null) {
+      Iterator<Annotation> iter = annotations.iterator();
+      while (iter.hasNext()) {
+        Annotation annotation = iter.next();
+        for (String defaultLogServiceKey : DEFAULT_LOG_LEVEL_KEYS) {
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java:687]: nit: for clarity, should this variable be `defaultLogLevelKey`?

File: java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java [hunk 3]
```diff
@@ -674,12 +676,88 @@ public static List<String> getDefaultLogMessageKeys() {
...
+          if (annotation.getKey().equals(defaultLogServiceKey)) {
+            iter.remove();
+            logLvlStr = annotation.getValue();
+            break;
+          }
+        }
+      }
+
+      if (logLvlStr == null && customLogLevelTags != null) {
+        // iterate over the set of custom service tags, breaking when one is found
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java:697]: nit: iterate over custom log level tags

File: java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java [hunk 4]
```diff
@@ -674,12 +676,88 @@ public static List<String> getDefaultLogMessageKeys() {
...
+
+  @Nullable
+  public static String getLogException(@Nullable List<Annotation> annotations,
+                                       @Nullable List<String> customLogExceptionTags) {
+    String exceptionStr = null;
+    if (annotations != null) {
+      Iterator<Annotation> iter = annotations.iterator();
+      while (iter.hasNext()) {
+        Annotation annotation = iter.next();
+        for (String defaultLogServiceKey : DEFAULT_LOG_EXCEPTION_KEYS) {
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java:724]: nit: ditto, name `defaultLogExceptionKey`

File: java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java [hunk 5]
```diff
@@ -674,12 +676,88 @@ public static List<String> getDefaultLogMessageKeys() {
...
+          if (annotation.getKey().equals(defaultLogServiceKey)) {
+            iter.remove();
+            exceptionStr = annotation.getValue();
+            break;
+          }
+        }
+      }
+
+      if (exceptionStr == null && customLogExceptionTags != null) {
+        // iterate over the set of custom service tags, breaking when one is found
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java:734]: nit: iterate over custom log exception tags

**Review by joannak-vmware — DISMISSED** [review]
Left a few nit comments. Otherwise, LGTM.

File: java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java (hunk 3 above)
Comment (panjwanipr-vmware) [reply java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java:697]: Done

File: java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java (hunk 5 above)
Comment (panjwanipr-vmware) [reply java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java:734]: Done

File: java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java (hunk 2 above)
Comment (panjwanipr-vmware) [reply java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java:687]: Done

**Review by joannak-vmware — APPROVED** [review]
LGTM

**Pull request merged**